packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -135,30 +135,30 @@ class VirtualGatewayListenerImpl extends VirtualGatewayListener {
    * mutual exclusivity
    */
   public bind(scope: Construct): VirtualGatewayListenerConfig {
-    const tlsConfig = this.tlsCertificate?.bind(scope);
     return {
       listener: {
         portMapping: {
           port: this.port,
           protocol: this.protocol,
         },
         healthCheck: this.healthCheck?.bind(scope, { defaultPort: this.port }).virtualGatewayHealthCheck,
-        tls: tlsConfig ? renderTls(tlsConfig) : undefined,
+        tls: renderTls(scope, this.tls),
```

Comment:
nit: do the `undefined` check here

Reply:
Undefined check is actually happening inside the `renderTls` function.

```typescript
function renderTls(scope: Construct, tls: TlsListener | undefined): CfnVirtualGateway.VirtualGatewayListenerTlsProperty | undefined {
  return tls
    ? {
      certificate: tls.certificate.bind(scope).tlsCertificate,
      mode: tls.mode,
    }
    : undefined;
}
```